opensquirrel/circuit.py

```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```

Comment:
This allows us to use the newer forms of type hinting in the older python versions. So, we can use `dict[int, int]` instead of `Dict[int, int]` or the `Mapping` from `collections.abc` instead of from `typing`. Both of which will eventualy be deprecated ([see](https://docs.python.org/3/library/typing.html#deprecated-aliases)).

Lastly, we can also use `int | float` instead of `Union[int, float]`, which (imo) reads much nicers, especcialy when you have complicated type hints, like `dict[str, str | list[str]] | Graph` instead of `Union[Dict[str, Union[str, List[str]], Graph]`